File: tests/fixtures/codegen/expected/typescript/runtime/queries/events/DeleteFinishedEvents.ts
import type { DeleteFinishedEventsParams } from "../../../types/queries/events/DeleteFinishedEvents.js";
export type { DeleteFinishedEventsParams } from "../../../types/queries/events/DeleteFinishedEvents.js";
import type * as pgsid from "./pgsid/queryable.js";
export const deleteFinishedEventsSql = "DELETE FROM public.events\nWHERE state = $1;";
export async function deleteFinishedEvents(db: pgsid.Queryable, params: DeleteFinishedEventsParams): Promise<number> {
    const result = await db.query(deleteFinishedEventsSql, [params["state"]]);
    return result.rowCount ?? 0;
}

File: tests/fixtures/codegen/expected/go/queries/events/deletefinishedevents.go
package events

import (
	"context"
	public "example.com/pgsid-fixture/generated/go/schema/public"
)

const DeleteFinishedEventsSQL = "DELETE FROM public.events\nWHERE state = $1;"

type DeleteFinishedEventsParams struct {
	State *public.EventState `db:"state"`
}

func (q *Queries) DeleteFinishedEvents(ctx context.Context, params DeleteFinishedEventsParams) (int64, error) {
	tag, err := q.db.Exec(ctx, DeleteFinishedEventsSQL, params.State)
	if err != nil {
		return 0, err
	}
	return tag.RowsAffected(), nil
}
